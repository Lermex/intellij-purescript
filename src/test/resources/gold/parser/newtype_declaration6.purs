module A where

newtype User9 a b = User9 a {
  name :: a -> b
}