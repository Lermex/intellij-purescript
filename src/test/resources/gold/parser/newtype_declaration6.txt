Purescript File
  PSProgramImpl(Program)
    PSModuleImpl(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSModuleNameImpl(ModuleName)
        PSQualifiedImpl(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSNewTypeDeclarationImpl(NewtypeDeclaration)
        PsiElement(newtype)('newtype')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User9')
            PsiElement(whitespace)(' ')
        PSTypeArgsImpl(TypeArgs)
          PSIdentifierImpl(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
          PSIdentifierImpl(GenericIdentifier)
            PsiElement(identifier)('b')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSTypeConstructorImpl(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User9')
            PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)('\n ')
        PSTypeAtomImpl(TypeAtom)
          PSObjectTypeImpl(ObjectType)
            PsiElement({)('{')
            PsiElement(whitespace)('\n  ')
            PSRowImpl(Row)
              PSIdentifierImpl(GenericIdentifier)
                PSIdentifierImpl(identifier)
                  PsiElement(identifier)('name')
                PsiElement(whitespace)(' ')
              PsiElement(::)('::')
              PsiElement(whitespace)(' ')
              PSTypeImpl(Type)
                PSTypeAtomImpl(TypeAtom)
                  PSIdentifierImpl(GenericIdentifier)
                    PsiElement(identifier)('a')
                    PsiElement(whitespace)(' ')
                PsiElement(->)('->')
                PsiElement(whitespace)(' ')
                PSTypeImpl(Type)
                  PSTypeAtomImpl(TypeAtom)
                    PSIdentifierImpl(GenericIdentifier)
                      PsiElement(identifier)('b')
                      PsiElement(whitespace)('\n ')
            PsiElement(})('}')