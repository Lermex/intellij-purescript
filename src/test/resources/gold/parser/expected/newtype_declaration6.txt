Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(Qualified)
        PSASTWrapperElement(ModuleName)
          PsiElement(proper name)('A')
      PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PsiElement(newtype)('newtype')
      PsiElement(whitespace)(' ')
      PsiElement(proper name)('User9')
      PsiElement(whitespace)(' ')
      PsiElement(identifier)('a')
      PsiElement(whitespace)(' ')
      PsiErrorElement:Expecting =
        PsiElement(identifier)('b')
        PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PsiElement(proper name)('User9')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PsiElement({)('{')
        PsiElement(whitespace)('\n  ')
        PsiElement(identifier)('name')
        PsiElement(whitespace)(' ')
        PsiElement(::)('::')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('a')
        PsiElement(whitespace)(' ')
        PsiElement(->)('->')
        PsiElement(whitespace)(' ')
        PsiElement(identifier)('b')
        PsiElement(whitespace)('\n')
  PsiErrorElement:Unexpected }. Expecting Module
    PsiElement(})('}')